runelite-client/src/main/java/net/runelite/client/game/chatbox/ChatboxPanelManager.java

```diff
@@ -0,0 +1,191 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.game.chatbox;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import java.util.Arrays;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.ScriptID;
+import net.runelite.api.VarClientInt;
+import net.runelite.api.events.ScriptCallbackEvent;
+import net.runelite.api.widgets.JavaScriptCallback;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.input.KeyListener;
+import net.runelite.client.input.KeyManager;
+import net.runelite.client.input.MouseListener;
+import net.runelite.client.input.MouseManager;
+import net.runelite.client.input.MouseWheelListener;
+
+@Singleton
+@Slf4j
+public class ChatboxPanelManager
+{
+	private final Client client;
+	private final ClientThread clientThread;
+	private final EventBus eventBus;
+
+	private final KeyManager keyManager;
+	private final MouseManager mouseManager;
+
+	private final Provider<ChatboxTextMenuInput> chatboxTextMenuInputProvider;
+	private final Provider<ChatboxTextInput> chatboxTextInputProvider;
+
+	@Getter
+	private ChatboxInput currentInput = null;
+
+	@Inject
+	private ChatboxPanelManager(EventBus eventBus, Client client, ClientThread clientThread,
+		KeyManager keyManager, MouseManager mouseManager,
+		Provider<ChatboxTextMenuInput> chatboxTextMenuInputProvider, Provider<ChatboxTextInput> chatboxTextInputProvider)
+	{
+		this.client = client;
+		this.clientThread = clientThread;
+		this.eventBus = eventBus;
+
+		this.keyManager = keyManager;
+		this.mouseManager = mouseManager;
+
+		this.chatboxTextMenuInputProvider = chatboxTextMenuInputProvider;
+		this.chatboxTextInputProvider = chatboxTextInputProvider;
+
+		eventBus.register(this);
```

Comment:
You should register this class to EventBus from `RuneLite` like we register every other subscriber

Reply:
How does this help? We do this in multiple other places, why is it not ok here?

Reply:
Its nice to have overview what is registered to EventBus in 1 class, and majority of stuff registered in EventBus is in RuneLite. Also, classes self-registering to EventBus are annoying because you need to actually inject the class somewhere so it's constructor will be called so they will be registered to EventBus.